spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapPropertySourceLocator.java

```diff
@@ -139,7 +140,8 @@ private void addPropertySourceIfNeeded(Function<String, Map<String, Object>> con
 			LOG.warn("Property source: " + name + "will be ignored because no properties could be found");
 		}
 		else {
-			composite.addFirstPropertySource(new MapPropertySource(name, map));
+			LOG.debug("will add file-based property source : " + name);
+			composite.addFirstPropertySource(new MountConfigMapPropertySource(name, map));
```

Comment:
this is 1/2 the fix, provide a well known class so that we can later check for it.